oxd-server/src/main/java/org/gluu/oxd/server/op/GetRequestUriOperation.java

```diff
@@ -0,0 +1,153 @@
+                    SignatureAlgorithm.fromString(rp.getRequestObjectSigningAlg());
+
+            if (algo == null) {
+                LOG.trace("The `request_object_signing_alg` parameter is not set. Using `none` algorithm by default.");
+                algo = SignatureAlgorithm.fromString("none");
+            }
+
+            Jwt jwt = setRequestObject(algo, rp, params);
+
+            //signing request object
+            String encodedSignature = getKeyGeneratorService().sign(jwt.getSigningInput(), jwt.getHeader().getKeyId(), rp.getClientSecret(), algo);
+            LOG.trace("encodedSignature : {} ", encodedSignature);
+            jwt.setEncodedSignature(encodedSignature);
+
+            //setting request object in Expired Object
+            getStateService().deleteExpiredObjectsByKey(rp.getOxdId());
+            getStateService().putRequestObject(rp.getOxdId(), jwt.toString());
+
+            String requestUri = baseOxdUrl(params.getOxdHostUrl(), params.getOxdId());
+            LOG.trace("RequestObject created successfully. request_uri : {} ", requestUri);
+
+            GetRequestUriResponse response = new GetRequestUriResponse();
+            response.setRequestUri(requestUri);
+            return response;
+
+        } catch (HttpException e) {
+            throw e;
+        } catch (Exception e) {
+            LOG.error("Error in creating `request_uri` response ", e);
+        }
+        throw new HttpException(ErrorResponseCode.FAILED_TO_GET_REQUEST_URI);
+    }
+
+    public Jwt setRequestObject(SignatureAlgorithm algo, Rp rp, GetRequestUriParams params) {
+        Jwt jwt = new Jwt();
+        //set header
+        jwt.getHeader().setType(JwtType.JWT);
+        try {
+            jwt.getHeader().setAlgorithm(algo);
+            String keyId = getKeyGeneratorService().getKeyId(Algorithm.fromString(algo.getName()), Use.SIGNATURE);
+            if (keyId != null) {
+                jwt.getHeader().setKeyId(keyId);
+            }
+        } catch (Exception e) {
+            LOG.error("Error in generating key Id.", e);
+        }
+        //set default claims
+        jwt.getClaims().setIssuer(rp.getClientId());
+        jwt.getClaims().setAudience(rp.getOpHost());
+        jwt.getClaims().setJwtId(UUID.randomUUID().toString());
+        jwt.getClaims().setClaim("client_id", rp.getClientId());
+        jwt.getClaims().setIssuedAt(new Date());
+        jwt.getClaims().setExpirationTime(Utils.addTimeToDate(new Date(), getConfigurationService().getConfiguration().getRequestObjectExpirationInMinutes(), Calendar.MINUTE));
+        jwt.getClaims().setClaim("response_type", rp.getResponseTypes());
+        //set claims from params
+        if (params.getParams() != null && !params.getParams().isEmpty()) {
+
+            Map<String, Object> claims = params.getParams();
+            claims.forEach((key, value) -> {
```

Comment:
Next 20 lines seems to be redundant, see `org.gluu.oxauth.model.jwt.JwtClaimSet#setClaimObject`.

Reply:
Done